### DISCUSSION TITLE
Should RemoteMessages support properties map per default?

### DISCUSSION THREAD

**Discussion by juliapampus** [eclipse-edc/Connector#2730] [Ideas]
Currently, e.g., the `TransferRequestMessage` has a properties map to allow sending custom key-value-pairs via the old ids protocol.

The dataspace protocol does not define extensible properties, so I would remove this map. However, with json-ld, it would be possible to provide this for any message type. For instance, if a dataspace would define its own profile of `dspace`, the EDC would automatically support it.

Approach would be to move the properties to the interface:

```java
public interface RemoteMessage {

    /**
     * Returns the transport protocol this message must be sent over.
     */
    String getProtocol();

    /**
     * Returns the recipient's connector address.
     */
    String getConnectorAddress();

    /**
     * Allows for values of custom namespaces.
     */
   Map<String, Object> getProperties();

}
```

The serialization output of
```java
Map.of("foo", "bar")
```
should be
```json
{
   "foo": "bar"
}
```

not 
```json
{
   "properties": [
      {
         "foo": "bar"
      }
   ]
}
```

**Comment by jimmarino** [comment]
Yes, we should support that since the DSpace protocol is built on JSON-LD and embraces its extensibility. This also means at the protocol layer, the message should be transformed into a JSON-LD structure using the JSON-P API and compaction performed before sending the serialized form on the wire.
